pymchelper/utils/mcscripter.py

```diff
@@ -0,0 +1,303 @@
+
+        _files = []
+        for item in self.c_dict["SYMLINKS"].split(","):
+            _files.append(item.strip())
+        self.c_dict["SYMLINKS"] = _files
+
+
+class McFile():
+    """
+    General MC single file object.
+    This will be used for the template files as well as the generated output files.
+    """
+    def __init__(self):
+        self.fname = ""  # filename
+        self.path = ""   # full path to this file (may be relative)
+        self.lines = []  # list of lines inside this file
+        self.symlink = False  # marker if file is a symlink
+        self.tdir = ""  # template directory
+
+    def write(self):
+        """
+        Write self to disk, create symlink if that is the case.
+        """
+        # check if target directory exists, create it, if not.
+        try:
+            os.makedirs(os.path.dirname(self.path))
+        except OSError as e:  # when python 2.7 is dropped this can be replaces tih FileExistsError: pass
+            if e.errno == errno.EEXIST:
+                pass  # silently accept existing directories
+            else:
+                logger.error('Directory not created.')
+                raise
+
+        if self.symlink:
+            link_file = os.path.join(self.tdir, self.fname)
+            link_target = os.path.join(self.path)
+            link_name = os.path.relpath(link_file, os.path.dirname(link_target))
+
+            try:
+                os.symlink(link_name, link_target)
+            except OSError as e:  # when python 2.7 is dropped this can be replaces tih FileExistsError: pass
+                if e.errno == errno.EEXIST:
+                    pass  # silently accept existing symlinks
+                else:
+                    logger.error('Symlink not created.')
+                    raise
+        else:
+            with open(self.path, 'w') as _f:
+                _f.writelines(self.lines)
+
+
+class Template():
+    """
+    Read all files and symlinks specified in the config file, and place them in a list of McFile objects.
+    """
+    def __init__(self, cfg):
+        self.files = []
+        self.read(cfg)
+
+    def read(self, cfg):
+        """
+        Reads all template files, and creates a list of McFile objects in self.files.
+        """
+
+        fname_list = cfg.c_dict["FILES"] + cfg.c_dict["SYMLINKS"]
+
+        for fname in fname_list:
+            tf = McFile()
+            tf.fname = fname
+            tf.path = os.path.join(cfg.c_dict["TDIR"], fname)
+            tf.tdir = cfg.c_dict["TDIR"]
+
+            with open(tf.path) as _f:
+                tf.lines = _f.readlines()
+
+            if fname in cfg.c_dict["SYMLINKS"]:
+                tf.symlink = True
+            else:
+                tf.symlink = False
+            self.files.append(tf)
+
+
+class Generator():
+    """
+    This generates and writes the output files based on the loaded template files and the config file.
+    """
+    def __init__(self, t, cfg):
```

Comment:
Better to call the parameters `tmpl` or `templ`. It is hard to guess what `t` stands for

Reply:
fixed also the others